amqp/src/main/scala/akka/stream/alpakka/amqp/AmqpConnector.scala

```diff
@@ -17,21 +17,20 @@ private[amqp] trait AmqpConnectorLogic { this: GraphStageLogic =>
   private var connection: Connection = _
   protected var channel: Channel = _
 
+  protected lazy val shutdownCallback = getAsyncCallback(onFailure)
```

Comment:
@juanjoDiaz I've followed your lead in your PR (#886) and have replaced my case class/tuple with a class-level shutdown listener and shutdown callback. I think this keeps the scope of change as small as possible while still accomplishing what the ticket reports.